src/PolicyEditor.js

```diff
@@ -496,6 +535,116 @@ export class PolicyEditor extends Component {
                             </Row>
                         </Accordion.Body>
                     </Accordion.Item>
+                    <Accordion.Item eventKey="snapshot-actions">
+                        <Accordion.Header><FontAwesomeIcon icon={faFileAlt} />&nbsp;Snapshot Actions</Accordion.Header>
+                        <Accordion.Body>
+                            <SectionHeaderRow />
+                            <Row>
+                                <LabelColumn name="Before Snapshot" help="Script to run before snapshot." />
+                                <WideValueColumn>{OptionalFieldNoLabel(this, "", "policy.actions.beforeSnapshotRoot.path", {})}</WideValueColumn>
+                                {EffectiveValue(this, "actions.beforeSnapshotRoot.path")}
+                            </Row>
+                            <Row>
+                                <LabelColumn name="Timeout" help="Timeout in seconds before Kopia kills the process." />
+                                <WideValueColumn>{OptionalNumberField(this, "", "policy.actions.beforeSnapshotRoot.timeout", {})}</WideValueColumn>
+                                {EffectiveValue(this, "actions.beforeSnapshotRoot.timeout")}
+                            </Row>
+                            <Row>
+                                <LabelColumn name="Command Mode" help="essential (must succeed, default behavior), optional (failures are tolerated) or async (kopia will start the action but not wait for it to finish)." />
```

Comment:
This is great! I only have few nits:

Since per-action fields are all the same, can you extract a helper function which gets a prefix field name and returns HTML fragment ?

Reply:
I'm afraid I've never written a line of React code in my live - I have some commented out/FIXME code in the latest commit based on what I fuzzily deduct from the existing patterns in that file, but that's not working. Help is appreciated...

Reply:
Ok, took another half hour and finally figured that one out, too.